src/main/templates/io/lettuce/core/api/RedisSortedSetCommands.java

```diff
@@ -1263,4 +1263,80 @@ Long zrevrangebyscoreWithScores(ScoredValueStreamingChannel<V> channel, K key, S
      */
     Long zunionstore(K destination, ZStoreArgs storeArgs, K... keys);
 
+    /**
+     * Add multiple sorted sets and returns the resulting sorted set.
+     *
+     * @param numkey the numeric key.
+     * @param keys the keys.
+     * @return List&lt;V&gt; array-reply list of elements.
+     */
+    List<V> zunion(long numkey, K... keys);
+
+    /**
+     * Add multiple sorted sets and returns the resulting sorted set.
+     *
+     * @param numkey the numeric key.
+     * @param storeArgs the storeArgs.
+     * @param keys the keys.
+     * @return List&lt;V&gt; array-reply list of elements.
+     */
+    List<V> zunion(long numkey, ZStoreArgs storeArgs, K... keys);
```

Comment:
My only concern on this - naming issue. We reuse similar ZStoreArgs but these commands even do not have "store" in their names. Shall we introduce just a wrapper class with new name which only extends it?

Reply:
Reuse `ZStoreArgs` is fine to me. : ) . @mp911de Can you share you opinions for this?

Reply:
`ZStoreArgs` doesn't really fit the new commands. While it was fine when we only had `Z…STORE` commands, the new variants require something better.

I'd suggest introducing `ZAggregateArgs` and retrofitting `ZStoreArgs` so that `ZStoreArgs extends ZAggregateArgs`. We don't need to deprecate anything. Related, introducing `zunionstore(…, ZAggregateArgs, …)` doesn't provide more value over `zunionstore(…, ZStoreArgs, …)`, therefore we should keep the existing methods as-is.

Reply:
I do personally like the approach with introducing `ZAggregateArgs` where `ZStoreArgs extends ZAggregateArgs`.
But, unfortunately, we still have to ship to `ZStoreArgs` all builder methods (since they are statically dispatched) and override all setter methods to return `ZStoreArgs`. With current PR code, this code would not compile:
```java
ZStoreArgs args = ZStoreArgs.Builder.max()
```